src/main/java/org/brapi/test/BrAPITestServer/service/germ/GermplasmService.java

```diff
@@ -677,14 +771,48 @@ private void updateSynonymEntities(List<GermplasmNewRequestSynonyms> synonyms, G
 		}
 	}
 
+	public List<GermplasmEntity> findByNames(List<String> germplasmNames) {
+		return germplasmRepository.findByGermplasmNameIn(germplasmNames);
+	}
+
+	public List<GermplasmEntity> findByIds(List<String> germplasmDbIds)
+		throws BrAPIServerException {
+		List<GermplasmEntity> germsFoundInDb = germplasmRepository.findByIdIn(germplasmDbIds);
+
+		Set<String> germIdsFoundInDB = germsFoundInDb.stream()
+				.map(BrAPIBaseEntity::getId)
+				.collect(Collectors.toSet());
+
+		if (!germIdsFoundInDB.containsAll(germplasmDbIds)) {
+			throw new BrAPIServerDbIdNotFoundException("Germplasm Ids passed to findByIds were not found in the DB", HttpStatus.BAD_REQUEST);
+		}
+
+		return germsFoundInDb;
+	}
+
+	// TODO: Add lookupType param to RQ Germplasm which can short-circuit all the lookup logic to only one query here.
 	public GermplasmEntity findByUnknownIdentity(String germplasmStr)
```

Comment:
This isn't used to search/query germplasm directly, we need to make a coordinated change in bi-api and the BrAPI server at some point, see https://breedinginsight.atlassian.net/browse/BI-2588.
```suggestion
	// TODO: Remove method once https://breedinginsight.atlassian.net/browse/BI-2588 is completed.
	public GermplasmEntity findByUnknownIdentity(String germplasmStr)
```

Reply:
To me this seems like a colliding of generality for the prod/test server vs BI's use case, but I think I'll let @BrapiCoordinatorSelby decide.

I like my comment here bc if we did implement it that way you guys can specify exactly how you want to search germs, and so could everybody else.

The ticket you referenced seems a bit different.  It sounds like you guys just want to have two totally different posts, one for germs first and then pedigree which can post based off of germs provided.  This seems doable, but likely slower than this implementation for time, data, and processing over the wire.

Reply:
The BrAPI spec and documentation do not require or indicate that pedigree connections will be created when you POST germplasm. It was an implementation choice (to support DeltaBreed I believe 😆) but it would be more reliable and likely more performant to change as described in [BI-2588](https://breedinginsight.atlassian.net/browse/BI-2588).

One issue is the current behavior breaks if a germplasm name used in the pedigree string has the "/" character in it, which is not unrealistic. I also think performance would be much better, but I haven't actually tested.

I worry that adding a `lookupType` param would encourage clients to rely on this behavior and I think we've proven it scales poorly.

Reply:
This shouldn't stand in the way of merging, it's just a comment.